ports/gstplugin/servosrc.rs

```diff
@@ -511,6 +533,18 @@ impl BaseSrcImpl for ServoGstSrc {
             .ok_or_else(|| gst_loggable_error!(CATEGORY, "Failed to get video info"))?;
         *self.info.lock().unwrap() = Some(info);
 
+        // Save the framerate if it is set
+        let framerate = outcaps
+            .get_structure(0)
+            .and_then(|cap| cap.get::<Fraction>("framerate"));
+        if let Some(framerate) = framerate {
+            let frame_duration_micros =
+                1_000_000 * *framerate.denom() as u64 / *framerate.numer() as u64;
+            debug!("Setting frame duration to {}micros", frame_duration_micros);
+            self.frame_duration_micros
```

Comment:
This never changes: Does it need to be an atomic?

Reply:
Renegotiating caps can happen at any time, and the src needs to be `Sync`, so yes it needs to be an `Atomic`.